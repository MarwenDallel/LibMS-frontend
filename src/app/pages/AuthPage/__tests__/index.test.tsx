import * as React from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { AuthPage } from '..';

const renderWithRouter = (component: JSX.Element) => {
  const history = createMemoryHistory();
  return {
    ...render(<Router history={history}>{component}</Router>),
  };
};

describe('<AuthPage  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = renderWithRouter(<AuthPage />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
