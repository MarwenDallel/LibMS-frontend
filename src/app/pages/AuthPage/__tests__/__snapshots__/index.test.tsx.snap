// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthPage  /> should match snapshot 1`] = `
.c2 {
  background-color: #007ea4;
}

.c1 {
  background-color: #203549;
}

.c0 {
  background-color: #f6f8fb;
}

<div
  class="c0 d-flex flex-column min-vh-100"
>
  <div
    class=""
  >
    <nav
      class="c1 navbar navbar-expand navbar-light"
    >
      <div
        class="container"
      >
        <a
          href="/"
        >
          <img
            alt=""
            height="122"
            src="SMU-LOGO.png"
            width="263"
          />
        </a>
      </div>
    </nav>
    <nav
      class="c2 navbar navbar-expand-md navbar-dark navbar navbar-expand-lg navbar-light"
    >
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="justify-content-center justify-content-md-between container"
        >
          <div
            class="font-weight-light mr-0 text-white"
          >
            My SMU-Library Account
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div
    class="text-left wrapper flex-grow-1 container"
  >
    <div
      class="d-block mt-5 mb-5 font-weight-light"
      style="font-size: 2em;"
    >
      Login to borrow books and check your requests
    </div>
    <div
      class="d-block"
      style="font-size: 1.232em;"
    >
      <div
        class="row"
      >
        <div
          class="border-left col"
        >
          <h4>
            SMU Students, Faculty and Staff
          </h4>
          <div
            class="mt-1"
          >
            <a
              data-testid="login-link"
              href="/login"
            >
              <button
                class="btn btn-primary btn-sm"
                data-testid="login-button"
                type="button"
              >
                Log in
              </button>
            </a>
          </div>
        </div>
        <div
          class="border-left col"
        >
          <h4>
            Create new account
          </h4>
          <div
            class="mt-1"
          >
            <a
              data-testid="register-link"
              href="/register"
            >
              <button
                class="btn btn-primary btn-sm"
                data-testid="register-button"
                type="button"
              >
                Register
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="mt-5 mb-5 row"
      >
        <div
          class="border-left col"
        >
          <h4>
            Policies & Information
          </h4>
          <div>
            <small>
              <a
                href="#!"
              >
                <i
                  class="bi bi-link-45deg"
                />
                Who can access?
              </a>
            </small>
          </div>
          <div>
            <small>
              <a
                href="#!"
              >
                <i
                  class="bi bi-link-45deg"
                />
                SMU Library Regulations
              </a>
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
